### PR TITLE
feat: migration to workspaces

### PR DESCRIPTION
**Pull request by milancermak**
Introducing Scarb workspaces, separating each module to its own Scarb package.

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [x] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Currently, Alexandria is one big package, all modules merged into a single dependency. I think this 1) hurts adoption and 2) stifles development (for example because there's no way how to do breaking changes in a single package). 

## What is the new behavior?

This PR adopts a [workspace](https://docs.swmansion.com/scarb/docs/reference/workspaces) based approach and separates each module to a single package.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

Yes in the sense how packages are installed and imported, but no backwards incompatible code changes.

## Other information

There's no material code changes. Most of the PR is just moving files around to a new dir structure. There are some small changes here and there, but mostly are cosmetic (e.g. alphabetical ordering of imports).

### CODE REVIEW

**Comment by milancermak** [comment]
cc @mkaput

**Comment by LucasLvy** [comment]
can you rebase por favor

File: Scarb.toml [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+]
 name = "alexandria"
 version = "0.1.0"
-description = "Starknet utilities"
+description = "Community maintained Cairo and Starknet libraries"
 homepage = "https://github.com/keep-starknet-strange/alexandria/"
 
-[dependencies]
-starknet = ">=2.1.0"
-
```
Comment (mkaput) [inline Scarb.toml:9]: You can leave it here, and then use `starknet.workspace = true` in particular packages: https://docs.swmansion.com/scarb/docs/reference/workspaces.html#dependencies

File: src/storage/Scarb.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "alexandria_storage"
+version = "0.1.0"
+description = "Starknet storage utilities"
+homepage = "https://github.com/keep-starknet-strange/alexandria/tree/src/storage"
+
+[[target.starknet-contract]]
+
```
Comment (mkaput) [inline src/storage/Scarb.toml:8]:
```suggestion

```

I don't see any contracts here, nor I don't think you'll want to build ones from this package? And not making it `[lib]` makes this package unusable as a dependency: https://docs.swmansion.com/scarb/docs/reference/targets.html#library

**Review by mkaput — CHANGES_REQUESTED** [review]

File: Scarb.toml (hunk 1 above)
Comment (milancermak) [reply Scarb.toml:9]: Getting this error:

```
error: failed to parse manifest at `/Users/m/projects/my-cairo-forks/alexandria/Scarb.toml`

Caused by:
    this virtual manifest specifies a [dependencies] section, which is not allowed
```

File: src/storage/Scarb.toml (hunk 2 above)
Comment (milancermak) [reply src/storage/Scarb.toml:8]: Yes, correct on the contracts part, removing the `[[target.starknet-contract]]` declaration.

Re: `[lib]` - shouldn't this be present in all `Scarb.toml` files in all modules then?

Comment (mkaput) [reply src/storage/Scarb.toml:8]: > Re: `[lib]` - shouldn't this be present in all `Scarb.toml` files in all modules then?

`[lib]` is the default and is automatically "added" if you don't specify any target :)

File: Scarb.toml (hunk 1 above)
Comment (maciektr) [reply Scarb.toml:9]: Hi @milancermak!

This needs to be called `[workspace.dependencies]`.
https://docs.swmansion.com/scarb/docs/reference/workspaces.html#dependencies

Comment (milancermak) [reply Scarb.toml:9]: I'm dumb. Thanks.

Comment (maciektr) [reply Scarb.toml]: This is just confusing when you do not define the root package, but we mimic this distinction from Cargo manifest format :D

Comment (mkaput) [reply Scarb.toml]: I'm making a PR which add a hint how to fix this error FYI :D

**Comment by milancermak** [comment]
@LucasLvy All done amigo

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**